extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/Trigger.java

```diff
@@ -9,6 +9,17 @@
  */
 public interface Trigger {
 
+    int OVERDUE_GRACE_PERIOD_IN_SECONDS = 1;
+
+    enum TriggerState {
```

Comment:
Could pls describe what each state constant actually means?

Reply:
Yes I made a comment about that, for now it's mirroring Quartz TriggerState, but we could create something more explicit. I was thinking of: 
RUNNING, 
PAUSED,
LAST_EXECUTION_ERROR: Exception during last execution
ERROR: Scheduler error
OVERDUE: would include BLOCKED status from Quartz, and locally computed overdue state based on next execution time

Reply:
TBH I'm not convinced this would be superuseful. `RUNNING/PAUSED` basically copies the functionality of `Scheduler.isRunning()` and `Scheduler.isPaused(String)`. `LAST_EXECUTION_ERROR/ERROR` only covers the last "result" and in fact, an exception thrown from a job does not always have to be an error. And `OVERDUE` is disputable because in theory you can "compute" the overdue value from the  `Trigger.getNextFireTime()`.

Reply:
yes I agree there is some overlap between the status and the methods from the scheduler. I still think we should be able to now if there was some exception thrown in a task. taking only the last one allow at least to spot a problem if it's happening on every run, but maybe we can do better (emitting a cdi event and let the user decide what to do ?).

Reply:
@mkouba I think the isOverdue method is useful because next fire time can be null, and this method can check the delay.
but to be more consistant with the rest of the api, it might be better to place it in the scheduler rather than in the Trigger, what do you think ?
```
        @Override
        public Instant getNextFireTime() {
            Date nextFireTime = context.getTrigger().getNextFireTime();
            Date nextFireTime = trigger.getNextFireTime();
            return nextFireTime != null ? nextFireTime.toInstant() : null;
```

Reply:
> . Ok, so isOverdue() + configurable OVERDUE_GRACE_PERIOD_IN_SECONDS seems to be useful.

+1, if we can make things configurable. But I've a more basic questions for me to be able to try and fully understand what we are trying to do here: 

@rmanibus @mkouba what is our intentions with these states and how do we expect the user to use them?

Reply:
@machi1990 I will remove the state, the goal of this is just to be able see what task are in the scheduler, and to be able to evaluate the health of each job.

Reply:
@mkouba @machi1990 I modified the PR according to this discussion, can you have a look ?

Reply:
I see thanks for the explanation and the PR. 

I've small code suggestion and renaming of the configuration. I'll let @mkouba have a general overview. 

Can we also update the documentation on the feature?

Reply:
@machi1990 thank you I updated the PR according to the suggestions. I will have a look at the doc

Reply:
@machi1990 I updated the documentation, tell me if it's ok for you !